lib/core/src/Cardano/Pool/DB.hs

```diff
@@ -29,7 +33,17 @@ import Data.Word
     ( Word64 )
 
 -- | A Database interface for storing pool production in DB.
-data DBLayer m = DBLayer
+--
+-- To use it, you will need the NamedFieldPuns extension and wrap operations
+-- with @atomically@:
+--
+-- Example:
+-- >>> :set -XNamedFieldPuns
+-- >>> DBLayer{atomically,putPoolProduction} = db
+-- >>> atomically $ putPoolProduction blockHeader pool
+--
+-- This gives you the power to also run /multiple/ operations atomically.
```

Comment:
:+1: , I'd even add: atomically, in isolation, with consistence and durability :D (SQLite transactions are ACID).